chia/util/streamable.py

```diff
@@ -179,12 +231,14 @@ def streamable(cls: Any):
 
     parse_functions = []
     try:
-        fields = cls1.__annotations__  # pylint: disable=no-member
+        metadata = _ClassMetadata.from_dataclass(dataclass=cls1)
```

Comment:
Ah, got it, you're generalizing `fields` here. How would you feel about moving the introspection utilities and types (like `_ClassMetadata` and `unstreamed_field` to their own file? That might make it more clear what their intended interface is.

Reply:
There's not actually a reason to cache this metadata, since any function that uses this could be dynamically "written" at class-definition time. See how `PARSE_FUNCTIONS_FOR_STREAMABLE_CLASS` are built up. Not sure if `_chia_streamable` is used elsewhere or not, but this might obviate the need for it.